client/app/src/main/java/com/microsoft/research/karya/data/service/WorkerAPI.kt

```diff
@@ -9,59 +10,38 @@ import com.microsoft.research.karya.data.model.karya.modelsExtra.WorkerObject
 import com.microsoft.research.karya.data.remote.response.CreationCodeResponse
 import okhttp3.ResponseBody
 import retrofit2.Response
-import retrofit2.http.Body
-import retrofit2.http.GET
-import retrofit2.http.Header
-import retrofit2.http.POST
-import retrofit2.http.PUT
-import retrofit2.http.Path
+import retrofit2.http.*
 
 interface WorkerAPI {
-    @GET("/worker/cc/{creation_code}")
-    suspend fun checkCreationCode(@Path("creation_code") id: String): Response<CreationCodeResponse>
 
-    @PUT("/worker/phone-auth")
-    suspend fun sendOTP(@Body worker: JsonObject): Response<WorkerRecord>
-
-    @PUT("/worker/phone-auth?resend=true")
-    suspend fun resendOTP(@Body worker: JsonObject): Response<WorkerRecord>
-
-    @PUT("/worker/update/cc")
-    suspend fun updateWorkerUsingCreationCode(@Body worker: WorkerObject): Response<WorkerRecord>
-
-    @PUT("/worker/refresh_token")
-    suspend fun refreshIdToken(
-        @Header("auth-provider") authProvider: String,
-        @Header("id-token") idTokenHeader: String
+    @PUT("/worker/otp")
+    suspend fun getOrVerifyOTP(
+        @Header("access-code") accessCode: String,
+        @Header("phone-number") phoneNumber: String,
+        @Header("otp") otp: String,
+        @Query("action") action: String, //TODO: Make this an enum class
     ): Response<WorkerRecord>
 
-    @POST("/worker_language_skill")
-    fun registerSkill(
-        @Body skillObject: WorkerLanguageSkillObject,
-        @Header("auth-provider") authProvider: String,
-        @Header("id-token") idTokenHeader: String
-    ): Response<WorkerLanguageSkillRecord>
-
-    @PUT("/worker_language_skill/{id}")
-    fun updateSkill(
-        @Body skillObject: WorkerLanguageSkillObject,
-        @Header("auth-provider") authProvider: String,
-        @Header("id-token") idTokenHeader: String,
-        @Path("id") workerLanguageSkillId: String
-    ): Response<WorkerLanguageSkillRecord>
-
-    @POST("/db/updates-for-worker")
-    fun getUpdates(
-        @Header("auth-provider") authProvider: String,
-        @Header("id-token") idTokenHeader: String,
-        @Body worker: WorkerRecord
-    ): Response<JsonArray>
+    /*
+    * This API would be used before OTP to determine App language and check the validity of access code
+    * */
+    @GET("/getWorker")
+    suspend fun getWorkerUsingAccessCode(
+        @Header("access-code") accessCode: String
+    ): Response<JsonObject>
```

Comment:
Same here

Reply:
It would just be a JSONObject with single field

Reply:
Yeah, but looking at directly it is not easy to figure out what the field is. Also, using a `JsonObject` means that we won't know the type at compile time and we have to rely on runtime checks for it.

Reply:
Sure, I will add a data object